analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py

```diff
@@ -0,0 +1,350 @@
+#
+# -------------------------------------------------------------------------
+"""
+"""
+from distutils.version import StrictVersion
+from pathlib import Path
+import os
+import re
+import shlex
+import shutil
+import subprocess
+import xml.etree.ElementTree as ET
+
+from codechecker_common.logger import get_logger
+
+from codechecker_analyzer.env import extend, get_binary_in_path
+
+from .. import analyzer_base
+
+from .config_handler import CppcheckConfigHandler
+from .result_handler import CppcheckResultHandler
+
+from ..config_handler import CheckerState
+
+LOG = get_logger('analyzer.cppcheck')
+
+
+def parse_checkers(cppcheck_output):
+    """
+    Parse cppcheck checkers list.
+    """
+    checkers = []
+
+    tree = ET.ElementTree(ET.fromstring(cppcheck_output))
+    root = tree.getroot()
+    errors = root.find('errors')
+    for error in errors.findall('error'):
+        name = error.attrib.get('id')
+        if name:
+            name = "cppcheck-" + name
+        msg = error.attrib.get('msg')
+        # TODO: Check severity handling in cppcheck
+        # severity = error.attrib.get('severity')
+
+        # checkers.append((name, msg, severity))
+        checkers.append((name, msg))
+
+    return checkers
+
+
+def parse_version(cppcheck_output):
+    """
+    Parse cppcheck version output and return the version number.
+    """
+    version_re = re.compile(r'^Cppcheck (?P<version>[\d\.]+)')
+    match = version_re.match(cppcheck_output)
+    if match:
+        return StrictVersion(match.group('version'))
+
+    return None
+
+
+class Cppcheck(analyzer_base.SourceAnalyzer):
+    """
+    Constructs the clang tidy analyzer commands.
+    """
+
+    ANALYZER_NAME = 'cppcheck'
+
+    def add_checker_config(self, checker_cfg):
+        LOG.error("Not implemented yet")
+
+    def get_analyzer_mentioned_files(self, output):
+        """
+        Return a collection of files that were mentioned by the analyzer in
+        its standard outputs, which should be analyzer_stdout or
+        analyzer_stderr from a result handler.
+        """
+        pass
+
+    def construct_analyzer_cmd(self, result_handler):
+        """
+        Construct analyzer command for cppcheck.
+        """
+        try:
+            config = self.config_handler
+
+            analyzer_cmd = [config.analyzer_binary]
+
+            # Enable or disable checkers.
+            enabled_severity_levels = set()
+
+            # Cppcheck runs with all checkers enabled for the time being
+            # the unneded checkers are passed as suppressed checkers
+            analyzer_cmd.append('--enable=all')
+
+            if enabled_severity_levels:
+                analyzer_cmd.append('--enable=' +
+                                    ','.join(enabled_severity_levels))
+
+            for checker_name, value in config.checks().items():
+                if value[0] == CheckerState.disabled:
+                    if checker_name.startswith("cppcheck-"):
```

Comment:
If everything is right  it cannot be false, but... The checking is there because of the hard coded start pos 9 in the string slicing on the line below. On pytthon3.9 this could be refactored with the nicer removeprefix() method.

Reply:
Cleared up the TODO comment too